dask/dataframe/core.py

```diff
@@ -3348,14 +3348,23 @@ def cummin(self, axis=None, skipna=True, out=None):
             out=out,
         )
 
+    def _validate_condition(self, cond):
+        if isinstance(cond, (list, np.ndarray)):
```

Comment:
I know this check covers most practical cases, but we are probably safer checking if `cond` is a supported type:

```python
        if not (
            is_dask_collection(cond) or
            is_dataframe_like(cond) or
            is_series_like(cond) or
            is_index_like(cond)
        ):
```

I realize that this isn't very concise, but I'm not sure we have much of a choice. For example, I don't think the `np.ndarray` check will complain about a cupy array.

Reply:
Thanks @rjzamora, that's a good point. I didn't even know that we have `is_dask_collection` check.

Reply:
I applied your suggestion, how is it looking now?

Reply:
IMHO, this should be either a `staticmethod` (notice you're not using `self` for anything) or a general utility method outside of the class.

This comment applies as long as you're planning to call it while checking the input of other related methods (e.g., `where`), so that it can be reused. Otherwise probably it's not worth to have a method for this check, and it should be hardcoded in `where` or `mask` itself.